docs: update the rwx volume document (#699)
harvester/docs: docs/rancher/csi-driver.md

## RWX Volumes Support

:::caution

RWX volumes feature now only worked with the Storage Network. This is a strong restriction for the customer to use RWX volumes. We already have a issue to track the corresponding enhancement: [[ENHANCEMENT] Support various vlan for RWX volume on the guest cluster](https://github.com/harvester/harvester/issues/7218)

:::

### Prerequisites

- Harvester v1.4 or later is installed on the host cluster.

> - Create [Storage Network](../advanced/storagenetwork.md) and configure it on Harvester cluster.
> 
>   We need to reserve a range for the VMs of guest cluster. Please use **exclude** to reserve some IP addresses.
>   ![](/img/v1.5/rancher/configure-storage-network-01.png)

Reviewer: ```suggestion
- A [storage network](../advanced/storagenetwork.md) is configured on the Harvester cluster.

  Use **exclude** to reserve a range of IP addresses for the guest cluster virtual machines. 

  ![](/img/v1.5/rancher/configure-storage-network-01.png)
```

> - Enable Storage Network for RWX volume:
> 
>   To enable Storage Network for RWX volume, you need to goto Longhorn Dashboard. Then, go **General** -> **Storage Network for RWX Volume Enabled**. This is a checkbox, please enable it.
>   ![](/img/v1.5/rancher/enable-rwx-storage-network-01.png)

Reviewer: ```suggestion
- The **Storage Network for RWX Volume** setting on the embedded Longhorn UI is enabled. 

  Go to **General**, and then select **Storage Network for RWX Volume Enabled**.
  
  ![](/img/v1.5/rancher/enable-rwx-storage-network-01.png)
```

  harvester-networkfs-manager-xvkgp                       1/1     Running     4 (37m ago)    3h41m
  ```

> - Create Guest Cluster with two network interface, one is mgmt, another one is storage network.
> 
>   The NAD **default/vlan101** is used to storage network.
>   ![](/img/v1.5/rancher/create-guest-cluster-with-two-nics.png)

Reviewer: ```suggestion
- The guest cluster has two network interfaces (`mgmt` and the storage network).

  The NAD **default/vlan101** is used for the storage network.

  ![](/img/v1.5/rancher/create-guest-cluster-with-two-nics.png)
```

  harvester-networkfs-manager-xvkgp                       1/1     Running     4 (37m ago)    3h41m
  ```

> - The VM must have two interfaces. The first one is the default network interface for cluster/external networking. The second one must be in a network which can connect to the storage network.

Reviewer: ```suggestion
- The virtual machine must have two network interfaces: 1) the default network interface for intra-cluster communications and allowing access from the infrastructure network (external to the Harvester cluster); and 2) a network that can connect to the storage network.
```

Reviewer: `cluster/external networking` was a bit vague. Is this what you mean?

  - SUSE and OpenSUSE: `zypper install -y nfs-client`

> - Manually assign the IP on the storage network interface.
> 
>   We already reserved the IP range on the previous, you can use these IPs to assign to the storage network interface.

Reviewer: ```suggestion
- An IP is manually assigned to the storage network interface.

  You can assign any of the reserved IPs using the following commands:
  
```

  - SUSE and OpenSUSE: `zypper install -y nfs-client`

- An IP is manually assigned to the storage network interface.

  You can assign any of the reserved IPs using the following commands:
  
  ```
  $ ip link set <storage network nic> up

>   $ ip a add <reserved IP> dev <storage network nic>

Reviewer 2: This can lead to misunderstandings, because these commands are not persistent after a reboot.

Author: Thanks @dfaltum, 
I added the note with that. This should be a reminder to help people to know this.

Hi @jillian-maroket, could you help to review this part again?
Thanks!

  - SUSE and OpenSUSE: `zypper install -y nfs-client`

- An IP is manually assigned to the storage network interface.

  You can assign any of the reserved IPs using the following commands:
  
  ```
  $ ip link set <storage network nic> up
  $ ip a add <reserved IP> dev <storage network nic>
  ```

>   :::note
>   The above command did not persist the IP address after a reboot. To make the IP address persistent, you can add the IP address to the network configuration file. (Depends on your guest OS)
>   :::

Reviewer: ```suggestion
  :::info important
  
  An IP that is assigned using the given commands does not persist after a reboot. To make the IP persistent, you must add it to the network configuration file of your guest operating system.

  :::
```

Author: updated!
